### PR TITLE
Reworked current delete command for vendor delete

### PR DESCRIPTION
**Pull request by Jasperabez**
The command format is updated as per mentioned in #77
`delete v/1`

- Deleting multiple items was not implemented due to the complexity of testing
- The command is extended in such a way adding delete for events should be trivial

### CODE REVIEW

**Comment by Jasperabez** [comment]
no idea why my codecov no passing despite adding test cases that accounts for it

**Comment by C5hives** [comment]
You might want to open an issue for this PR just to follow the workflow requirements for the tP. Just admin work—the PR can remain as is.

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 1]
```diff
@@ -18,22 +19,46 @@ public class DeleteCommand extends Command {
...
     }
 
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
+
+        if (itemType == ItemTypeToDelete.VENDOR) {
+            return deleteVendor(model);
+        } else {
+            throw new CommandException(Messages.MESSAGE_INVALID_COMMAND_FORMAT);
```
Comment (C5hives) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:57]: This logic feels out of place and requires you to create another enum to perform checks.

You might want to consider shifting this logic to be handled by `DeleteCommandParser` instead. The parser itself already does `Prefix` checks, which you can take advantage of and forgo the enum `ItemTypeToDelete` entirely.

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java
Comment (C5hives) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:1]: The name DeleteCommand will no longer be descriptive enough once support for deleting vendors and events is added.

You might want to consider turning this class into an abstract class and use polymorphism to handle deleting vendors and events as separate classes.

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 2]
```diff
@@ -18,22 +19,46 @@ public class DeleteCommand extends Command {
...
 
+    // The index of item to be deleted
     private final Index targetIndex;
+    private final ItemTypeToDelete itemType;
 
-    public DeleteCommand(Index targetIndex) {
+    /**
+     * Creates a DeleteCommand to delete the specified item.
+     */
+    public DeleteCommand(ItemTypeToDelete itemType, Index targetIndex) {
```
Comment (C5hives) [inline src/main/java/seedu/address/logic/commands/DeleteCommand.java:45]: Passing flags around to determine class behaviour is a code smell. As mentioned, consider polymorphism instead.

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java [hunk 3]
```diff
@@ -18,22 +19,46 @@ public class DeleteCommand extends Command {
 
     public static final String COMMAND_WORD = "delete";
 
+    /**
+     * Enum to specify the type of item to delete.
+     */
+    public enum ItemTypeToDelete {
+        VENDOR,
+    }
```
[the remark this answers is not in the document]
Comment (C5hives) [reply src/main/java/seedu/address/logic/commands/DeleteCommand.java:27]: Following the subsequent comments, I would recommend getting rid of this enum entirely and rely on the existing `CliSyntax` class to perform checks.

**Review by C5hives — CHANGES_REQUESTED** [review]
Logic is generally ok, but some refactoring could be done to better adhere to OOP principles.

Let me know if you have any other suggestions instead.

File: src/main/java/seedu/address/logic/commands/DeleteCommand.java
Comment (Jasperabez) [reply src/main/java/seedu/address/logic/commands/DeleteCommand.java:1]: Refactored!

File: src/test/java/seedu/address/logic/commands/DeleteCommandTest.java [hunk 4]
```diff
@@ -1,120 +1,35 @@
...
-        assertFalse(deleteFirstCommand.equals(null));
-
-        // different vendor -> returns false
-        assertFalse(deleteFirstCommand.equals(deleteSecondCommand));
+        @Override
+        public CommandResult execute(Model model) {
+            // This method is not used in the test
+            return null;
+        }
     }
```
Comment (C5hives) [inline src/test/java/seedu/address/logic/commands/DeleteCommandTest.java:26]: You might want to put the stub at the bottom of the file and let the test methods come first.

The stub is usually not as important and placing it near the bottom of the file would help get to the meat of the class (which are the tests). Also just for standardisation across all the test classes.

**Review by C5hives — CHANGES_REQUESTED** [review]
Nice job on the refactoring! Just one more nitpick—the stub used in `DeleteCommandTest` could be moved to the bottom of the file to allow the test methods to come first

**Review by C5hives — APPROVED** [review]
Looks good!

**Pull request merged**